src/circuit/sapling/mod.rs

```diff
@@ -379,8 +428,32 @@ impl<'a, E: JubjubEngine> Circuit<E> for Spend<'a, E> {
             )?.get_x().clone(); // Injective encoding
         }
 
-        // Expose the anchor
-        cur.inputize(cs.namespace(|| "anchor"))?;
+        {
+            let real_anchor_value = self.anchor;
+
+            // Expose the real anchor. If `enforce` was true,
+            // this has to be equal to the result of the merkle
+            // path check (cur); if not, it can be anything.
+            let rt = num::AllocatedNum::alloc(
+                cs.namespace(|| "conditional anchor"),
+                || {
+                    Ok(*real_anchor_value.get()?)
+                }
+            )?;
+
+            // (cur - rt) * enforce = 0
```

Comment:
This could be just ``(cur - rt) × (value) = 0``, eliminating the ``enforce`` variable.